datafusion/sql/src/planner.rs

```diff
@@ -46,7 +46,7 @@ use crate::utils::make_decimal_type;
 /// functions referenced in SQL statements
 pub trait ContextProvider {
     /// Getter for a datasource
-    fn get_table_provider(&self, name: TableReference) -> Result<Arc<dyn TableSource>>;
+    fn get_table_source(&self, name: TableReference) -> Result<Arc<dyn TableSource>>;
```

Comment:
This would be a breaking API change -- to give people time to adjust as well as a hint on how to fix it, could you please leave a backwards compatible, deprecated version

Perhaps something like

```rust
#[deprecated(
    since = "32.0.0",
    note = "please use `get_table_source` instead"
)]
    fn get_table_provider(&self, name: TableReference) -> Result<Arc<dyn TableSource>> {
      self.get_table_source(name)
    }
```

Reply:
Sure, added!